pybamm/simulation.py

```diff
@@ -0,0 +1,236 @@
+import pybamm
+import numpy as np
+import copy
+
+
+class Simulation:
+    """A Simulation class for easy building and running of PyBaMM simulations.
+
+    Parameters
+    ----------
+    model : :class:`pybamm.BaseModel`
+        The model to be simulated
+    """
+
+    def __init__(self, model):
+        self.model = model
+        self.set_defaults()
+        self.reset()
+
+    def set_defaults(self):
+        """
+        A method to set all the simulation specs to default values for the
+        supplied model.
+        """
+        self.geometry = self._model.default_geometry
+        self._parameter_values = self._model.default_parameter_values
+        self._submesh_types = self._model.default_submesh_types
+        self._var_pts = self._model.default_var_pts
+        self._spatial_methods = self._model.default_spatial_methods
+        self._solver = self._model.default_solver
+        self._quick_plot_vars = None
+
+    def reset(self):
+        """
+        A method to reset a simulation back to its unprocessed state.
+        """
+        self.model = self._model_class(self._model_options)
+        self.geometry = copy.deepcopy(self._unprocessed_geometry)
+        self._mesh = None
+        self._disc = None
+        self._solution = None
+        self._status = "Unprocessed"
+
+    def parameterize(self):
```

Comment:
minor point, but I dont think I like the name `parameterize` -- I think it suggests it is finding the parameters for the model. maybe `set_parameters`?

Reply:
Agreed

Reply:
I agree